The PostCommit YAML Xlang Direct job is flaky

The PostCommit YAML Xlang Direct is failing over 50% of the time.
Please visit https://github.com/apache/beam/actions/workflows/beam_PostCommit_Yaml_Xlang_Direct.yml?query=is%3Afailure+branch%3Amaster to see all failed workflow runs.
See also Grafana statistics: http://metrics.beam.apache.org/d/CTYdoxP4z/ga-post-commits-status?orgId=1&viewPanel=11&var-Workflow=PostCommit%20YAML%20Xlang%20Direct

@tarun-google One test is still failing with _tempLocation_ NPE: `FAILED apache_beam/yaml/integration_tests.py::BigqueryTest::test_ReadFromBigQuery_InlineProvider_1`
Could you please take a look?

Majority of the test failure is due to test `apache_beam/yaml/integration_tests.py::EnrichmentTest::test_only`, where the matching row was not found.

```
E         File "/runner/_work/beam/beam/sdks/python/apache_beam/transforms/enrichment_handlers/bigquery.py", line 225, in __call__
E           raise ValueError(f"no matching row found for query: {query}")
E       ValueError: no matching row found for query: SELECT * FROM apache-beam-testing.yaml_bq_it__cd7eb5106be0459e977bd642fce4f024.tmp_table WHERE label = '37a'
```

For example:
- https://github.com/apache/beam/actions/runs/19643486196/job/56252848556
- https://github.com/apache/beam/actions/runs/19632792849/job/56216293159
- https://github.com/apache/beam/actions/runs/19624460698/job/56190629499

--- a/.github/trigger_files/beam_PostCommit_Yaml_Xlang_Direct.json
+++ b/.github/trigger_files/beam_PostCommit_Yaml_Xlang_Direct.json
@@ -1,4 +1,4 @@
 {
   "comment": "Modify this file in a trivial way to cause this test suite to run",
-  "revision": 3
+  "revision": 6
 }

--- a/.github/workflows/beam_PostCommit_Yaml_Xlang_Direct.yml
+++ b/.github/workflows/beam_PostCommit_Yaml_Xlang_Direct.yml
@@ -80,7 +80,7 @@ jobs:
       - name: run PostCommit Yaml Xlang Direct script
         uses: ./.github/actions/gradle-command-self-hosted-action
         with:
-          gradle-command: :sdks:python:postCommitYamlIntegrationTests -PyamlTestSet=${{ matrix.test_set }} -PbeamPythonExtra=ml_test,yaml
+          gradle-command: :sdks:python:postCommitYamlIntegrationTests -PyamlTestSet=${{ matrix.test_set }} -PbeamPythonExtra=p310_ml_test,yaml
       - name: Archive Python Test Results
         uses: actions/upload-artifact@v7
         if: failure()

